docker环境下初始环境-数据库初始化失败

docker logs gshark-server
您正在使用GVA_CONFIG环境变量,config的路径为config.docker.yaml
create log directory
[GShark]2026/06/12-06:51:40     info    use middleware logger
[GShark]2026/06/12-06:51:40     info    router register success
[GShark]2026/06/12-06:51:40     info    server run success on   {"address": ":8888"}
2026/06/12 06:51:40 1 :8888
[GShark]2026/06/12-06:52:34     info    前往初始化数据库
[GIN] 2026/06/12 - 06:52:34 | 200 |     154.883µs |     10.1.113.18 | POST     "/init/checkdb"

[Mysql] --> sys_users 表初始数据成功!

[Mysql] --> sys_apis 表初始数据成功!

2026/06/12 06:52:39 /go/src/github.com/madneal/gshark/server/source/authority_menu.go:14 Error 1146 (42S02): Table 'gshark.authority_menu' doesn't exist
[2.419ms] [rows:0] SELECT * FROM `authority_menu` WHERE `authority_menu`.`deleted_at` IS NULL

[Mysql] --> authority_menu 视图创建成功!

[Mysql] --> sys_authorities 表初始数据成功!

[Mysql] --> sys_authority_menus 表初始数据成功!

2026/06/12 06:52:39 /go/src/github.com/madneal/gshark/server/source/casbin.go:115 Error 1406 (22001): Data too long for column 'v1' at row 37
[2.083ms] [rows:0] INSERT INTO `casbin_rules` (`p_type`,`v0`,`v1`,`v2`,`v3`,`v4`,`v5`) VALUES ('p','888','/base/login','POST','','',''),('p','888','/user/register','POST','','',''),('p','888','/api/createApi','POST','','',''),('p','888','/api/getApiList','POST','','',''),('p','888','/api/getApiById','POST','','',''),('p','888','/api/deleteApi','POST','','',''),('p','888','/api/updateApi','POST','','',''),('p','888','/api/getAllApis','POST','','',''),('p','888','/authority/createAuthority','POST','','',''),('p','888','/authority/deleteAuthority','POST','','',''),('p','888','/authority/getAuthorityList','POST','','',''),('p','888','/authority/setDataAuthority','POST','','',''),('p','888','/authority/updateAuthority','PUT','','',''),('p','888','/authority/copyAuthority','POST','','',''),('p','888','/menu/getMenu','POST','','',''),('p','888','/menu/getMenuList','POST','','',''),('p','888','/menu/addBaseMenu','POST','','',''),('p','888','/menu/getBaseMenuTree','POST','','',''),('p','888','/menu/addMenuAuthority','POST','','',''),('p','888','/menu/getMenuAuthority','POST','','',''),('p','888','/menu/deleteBaseMenu','POST','','',''),('p','888','/menu/updateBaseMenu','POST','','',''),('p','888','/menu/getBaseMenuById','POST','','',''),('p','888','/user/changePassword','POST','','',''),('p','888','/user/getUserList','POST','','',''),('p','888','/user/setUserAuthority','POST','','',''),('p','888','/user/deleteUser','DELETE','','',''),('p','888','/casbin/updateCasbin','POST','','',''),('p','888','/casbin/getPolicyPathByAuthorityId','POST','','',''),('p','888','/casbin/casbinTest/:pathParam','GET','','',''),('p','888','/jwt/jsonInBlacklist','POST','','',''),('p','888','/system/getSystemConfig','POST','','',''),('p','888','/system/setSystemConfig','POST','','',''),('p','888','/system/getServerInfo','POST','','',''),('p','888','/system/emailTest','POST','','',''),('p','888','/system/botTest','GET','','',''),('p','888','/sysDictionaryDetail/createSysDictionaryDetail','POST','','',''),('p','888','/sysDictionaryDetail/deleteSysDictionaryDetail','DELETE','','',''),('p','888','/sysDictionaryDetail/updateSysDictionaryDetail','PUT','','',''),('p','888','/sysDictionaryDetail/findSysDictionaryDetail','GET','','',''),('p','888','/sysDictionaryDetail/getSysDictionaryDetailList','GET','','',''),('p','888','/sysDictionary/createSysDictionary','POST','','',''),('p','888','/sysDictionary/deleteSysDictionary','DELETE','','',''),('p','888','/sysDictionary/updateSysDictionary','PUT','','',''),('p','888','/sysDictionary/findSysDictionary','GET','','',''),('p','888','/sysDictionary/getSysDictionaryList','GET','','',''),('p','888','/sysOperationRecord/createSysOperationRecord','POST','','',''),('p','888','/sysOperationRecord/deleteSysOperationRecord','DELETE','','',''),('p','888','/sysOperationRecord/updateSysOperationRecord','PUT','','',''),('p','888','/sysOperationRecord/findSysOperationRecord','GET','','',''),('p','888','/sysOperationRecord/getSysOperationRecordList','GET','','',''),('p','888','/sysOperationRecord/deleteSysOperationRecordByIds','DELETE','','',''),('p','888','/user/setUserInfo','PUT','','',''),('p','888','/rule/deleteRule','DELETE','','',''),('p','888','/rule/createRule','POST','','',''),('p','888','/rule/deleteRuleByIds','DELETE','','',''),('p','888','/rule/updateRule','PUT','','',''),('p','888','/rule/findRule','GET','','',''),('p','888','/rule/getRuleList','GET','','',''),('p','888','/rule/switchRuleStatus','POST','','',''),('p','888','/rule/uploadRules','POST','','',''),('p','888','/token/createToken','POST','','',''),('p','888','/token/deleteToken','DELETE','','',''),('p','888','/token/deleteTokenByIds','DELETE','','',''),('p','888','/token/updateToken','PUT','','',''),('p','888','/token/findToken','GET','','',''),('p','888','/token/getTokenList','GET','','',''),('p','888','/searchResult/createSearchResult','POST','','',''),('p','888','/searchResult/deleteSearchResult','DELETE','','',''),('p','888','/searchResult/deleteSearchResultByIds','DELETE','','',''),('p','888','/searchResult/updateSearchResult','POST','','',''),('p','888','/searchResult/findSearchResult','GET','','',''),('p','888','/searchResult/getSearchResultList','GET','','',''),('p','888','/searchResult/exportSearchResult','GET','','',''),('p','888','/searchResult/updateSearchResultStatusByIds','POST','','',''),('p','888','/searchResult/getTaskStatus','GET','','',''),('p','888','/subdomain/createSubdomain','POST','','',''),('p','888','/subdomain/deleteSubdomain','DELETE','','',''),('p','888','/subdomain/deleteSubdomainByIds','DELETE','','',''),('p','888','/subdomain/updateSubdomain','PUT','','',''),('p','888','/subdomain/findSubdomain','GET','','',''),('p','888','/subdomain/getSubdomainList','GET','','',''),('p','888','/filter/createFilter','POST','','',''),('p','888','/filter/deleteFilter','DELETE','','',''),('p','888','/filter/deleteFilterByIds','DELETE','','',''),('p','888','/filter/updateFilter','PUT','','',''),('p','888','/filter/findFilter','GET','','',''),('p','888','/filter/getFilterList','GET','','',''),('p','888','/task/getTaskList','GET','','',''),('p','888','/task/createTask','POST','','',''),('p','888','/task/switchTaskStatus','POST','','','')
[GShark]2026/06/12-06:52:39     error   自动创建数据库失败      {"err": "Error 1406 (22001): Data too long for column 'v1' at row 37"}
DROP DATABASE IF EXISTS gshark;
[GIN] 2026/06/12 - 06:52:39 | 200 |  637.289804ms |     X.X.X.X | POST     "/init/initdb"



docker-compose up -d 后通过web 初次访问系统，点击初始化报告“数据库初始化失败，请查看后台日志”，通过docker logs 查看server容器后发现该问题。数据库连接正常，mysql容器也是要求版本8.0.21

--- a/server/service/sys_initdb.go
+++ b/server/service/sys_initdb.go
@@ -127,9 +127,9 @@ func InitDB(conf request.InitDB) error {
 	err = initDB(
 		source.Admin,
 		source.Api,
-		source.AuthorityMenu,
 		source.Authority,
 		source.AuthoritiesMenus,
+		source.AuthorityMenu,
 		source.Casbin,
 		source.DataAuthorities,
 		source.Dictionary,

--- a/server/source/casbin.go
+++ b/server/source/casbin.go
@@ -107,6 +107,7 @@ var carbines = []gormadapter.CasbinRule{
 
 func (c *casbin) Init() error {
 	global.GVA_DB.AutoMigrate(gormadapter.CasbinRule{})
+	global.GVA_DB.Exec("ALTER TABLE casbin_rule MODIFY COLUMN v1 VARCHAR(100)")
 	return global.GVA_DB.Transaction(func(tx *gorm.DB) error {
 		if tx.Find(&[]gormadapter.CasbinRule{}).RowsAffected > 1 {
 			color.Danger.Println("\n[Mysql] --> casbin_rule 表的初始数据已存在!")

